### PR TITLE
Fixed #28987 -- Fixed altering ManyToManyField when changing to self-referential.

### PR DESCRIPTION
**Pull request by DevilsAutumn**
[Ticket #28987](https://code.djangoproject.com/ticket/28987)

### CODE REVIEW

File: django/db/backends/sqlite3/schema.py [hunk 1]
```diff
@@ -236,6 +241,14 @@ def is_self_referential(f):
...
                     self.effective_default(create_field),
                 )
+        # Alter field pointing to the model itself.
+        if alter_self_field:
+            old_self_field, new_self_field = alter_self_field
+            body.pop(old_self_field.name, None)
+            mapping.pop(old_self_field.column, None)
+            body[new_self_field.name] = new_self_field
+            mapping[new_self_field.column] = self.quote_name(old_self_field.column)
+            rename_mapping[old_self_field.name] = new_self_field.name
```
Comment (felixxm) [inline django/db/backends/sqlite3/schema.py:251]: We're doing the same for `alter_field`. It's probably worth to reuse the same code for both, by changing the `alter_field` argument to `alter_fields` (`List[Tuple]`, it was a list before 767849b765e881ee905f2bc1175d1c1f47445554) or by:
```python
        # Add in any altered fields
        for alter_field in [alter_field, alter_self_field]:
            if alter_field:
                old_field, new_field = alter_field
                ...
```

File: django/db/backends/sqlite3/schema.py [hunk 2]
```diff
@@ -507,6 +520,14 @@ def _alter_many_to_many(self, model, old_field, new_field, strict):
                         new_field.m2m_reverse_field_name()
                     ),
                 ),
+                alter_self_field=(
```
Comment (felixxm) [inline django/db/backends/sqlite3/schema.py:523]: Have you seen the @andrewgodwin comment above? (see 03900a02d5fd2efd6ae98bdb9974b78146f63040)

> _The field that points to the target model is needed, so_
> _we can tell alter_field to change it - this is_
> _m2m_reverse_field_name() (**as opposed to m2m_field_name()**,_
> _which points to our model)._

I'm a little worried that we're missing something here :thinking: We should at least by update this comment.

**Review by felixxm — COMMENTED** [review]
@DevilsAutumn Thanks :+1:

File: django/db/backends/sqlite3/schema.py (hunk 1 above)
Comment (DevilsAutumn) [reply django/db/backends/sqlite3/schema.py:251]: Thanks for the review @felixxm .  ```alter_self_field``` is only to be used for altering a m2m self field. I'm afraid passing both in a single argument would create a problem for altering non-m2m field.

Comment (felixxm) [reply django/db/backends/sqlite3/schema.py:251]: I've checked and it works fine.

Comment (DevilsAutumn) [reply django/db/backends/sqlite3/schema.py:251]: Made changes according to second method 👍

Comment (felixxm) [reply django/db/backends/sqlite3/schema.py:251]: I don't see any changes.

Comment (DevilsAutumn) [reply django/db/backends/sqlite3/schema.py:251]: Haven't pushed yet,was waiting for your suggestions on the comment.

**Comment by DevilsAutumn** [comment]
@felixxm pushed the changes 👍

**Comment by DevilsAutumn** [comment]
@felixxm If you could please review again.🙇

File: django/db/backends/sqlite3/schema.py (hunk 1 above)
Comment (felixxm) [reply django/db/backends/sqlite3/schema.py]: As far as I'm aware, changing to `alter_fields` makes this API clearer. I pushed edits, please take a look.

Comment (DevilsAutumn) [reply django/db/backends/sqlite3/schema.py]: Thanks a lot! Looks better now.

**Comment by felixxm** [comment]
@DevilsAutumn Thanks :+1:

**Review by felixxm — APPROVED** [review]

**Pull request merged**